### DISCUSSION TITLE
4ch mains CT board for ESP8266

### DISCUSSION THREAD

**Discussion by satmanuk2** [arendst/Tasmota#18068] [Q&A]
has anyone tried the below board with tasmota ?  ive asked the creator & but he has no experience with tasmota, however it just uses the built in ADC & a 2x gpio multiplexer to select the CT to gather readings from, so i dont see why it wouldnt work.

https://github.com/Mottramlabs/4-Channel-Mains-Current-Sensor-ESP8266
https://www.mottramlabs.com/esp_products.html

**Comment by barbudor** [comment]
I see 2 boards with the same title
- One using the ESP's own ADC
- One using an additional ADS1115 ADC

The 2nd one would probably works directly as the ADS1115 is supported by Tasmota.
However you would need to handle the conversions from ADC readings (volts) into Current either in a rule or in your backend

The 1st one would be more tricky as Tasmota do not support natively an analog multiplexer in front of the ADC. It would be possible to use Relay outputs to drive the analog multiplexer but that would complexify the process for extracting the current.

I don't know if you have already purchased the board and the mini-D1 or ESP12 module that would go with but choosing an ESP32 based module (like ESP-C3-12F) or ESP32-mini-D1 would make easier to use Berry for that
Alternatively on ESP8266 you may be able to use Scripting-Language

Of course, it is also still possible to code a dedicated native driver if you feel so

Someone was also writing recently here about a board he is designing with ESP32 and CT-inputs, but I can't find again the Discussion

**Reply by barbudor** [reply]
I found that other discussion : https://github.com/arendst/Tasmota/discussions/17801
And the web site : https://ditronix.net/ (I'm not related with him or his business)
